### ISSUE TITLE
Test fine-tuning training with different precisions

### ISSUE THREAD

**Issue by gkumbhat** [caikit/caikit-nlp#78]
## Description

Target:
1. bf16 
2. ~~FSDP based fp16 fine tuning~~ [ TO BE CARRIED IN DIFFERENT STORY]
3. ~~DDP FP32 fine tuning~~ [

## Discussion

Provide detailed discussion here

## Acceptance Criteria

- [ ] Unit tests cover new/changed code
- [ ] Examples build against new/changed code
- [ ] READMEs are updated
- [ ] Type of [semantic version](https://semver.org/) change is identified

**Comment by olson-ibm** [comment]
Updated instruction, per @gkumbhat:

Model: Llama 7b
GPU: A100
Precision: bf16
Input Seq Length: 512
Output Seq Length: 256
Batch size: 2 

Get a quick-and-dirty test out with the above and report back.

**Comment by olson-ibm** [comment]
LLAMA2-7b successfully loaded for fine tuning using bfloat16 on a A100 80GB GPU. Fine tuning was not completed and aborted after 6 hours to give the GPU back. Estimate it will take 96 hours to complete the fine tuning.

Bugfix PR #135 was pushed to complete this.

**Comment by gkumbhat** [comment]
Thanks @olson-ibm . FSDP verification is currently blocked by #117 . Opened separate issue for this: https://github.com/caikit/caikit-nlp/issues/138

**Comment by olson-ibm** [comment]
Command used: 

```
python examples/run_fine_tuning.py --dataset "glue/rte"  --model_name $LLAMA_HOME --num_epochs 1 --output_dir $MODEL_OUTPUT_DIR --batch_size=2 --accumulate_steps 16 --max_source_length
 512 --max_target_length 256 --metric rouge --torch_dtype bfloat16
```

Execution benchmark:

```
{'train_runtime': 521.6131, 'train_samples_per_second': 14.321, 'train_steps_per_second': 0.447, 'train_loss': 1.5158040451901154, 'epoch': 1.0}
```
